src/api/SnykApiClient.ts

```diff
@@ -0,0 +1,474 @@
+import { TargetData } from "../types/targetsTypes";
+import { OrgData } from "../types/orgsTypes";
+import { ProjectsData } from "../types/projectsTypes";
+import {
+    SNYK_ANNOTATION_TARGETID,
+    SNYK_ANNOTATION_TARGETNAME,
+    SNYK_ANNOTATION_TARGETS,
+    SNYK_ANNOTATION_PROJECTIDS,
+    SNYK_ANNOTATION_EXCLUDE_PROJECTIDS,
+    SNYK_ANNOTATION_ORG,
+    SNYK_ANNOTATION_ORGS,
+} from "../config";
+import { mockedProjects } from "../utils/mockedProjects";
+import { mockedIssues } from "../utils/mockedIssues";
+import { Entity } from "@backstage/catalog-model";
+import { mockedDepGraphs } from "../utils/mockedDepGraphs";
+import { mockedProjectDetails } from "../utils/mockedProjectDetails";
+import { IssuesCount } from "../types/types";
+import { Issue } from "../types/unifiedIssuesTypes";
+const DEFAULT_PROXY_PATH_BASE = "";
+
+type Options = {
+    discoveryApi: DiscoveryApi;
+    /**
+     * Path to use for requests via the proxy, defaults to ''
+     */
+    proxyPathBase?: string;
+    configApi: ConfigApi;
+    fetchApi: FetchApi;
+};
+export const snykApiRef: ApiRef<SnykApi> = createApiRef<SnykApi>({
+    id: "plugin.snyk.service",
+});
+
+export interface SnykApi {
+    listAllAggregatedIssues(orgName: string, projectId: string): Promise<any>;
+
+    getProjectDetails(orgName: string, projectId: string): Promise<any>;
+
+    getCompleteProjectsListFromAnnotations(
+        orgId: string,
+        annotations: Record<string, string>,
+        ignoreMissingTargets: boolean
+    ): Promise<ProjectsData[]>;
+
+    getDependencyGraph(orgName: string, projectId: string): Promise<any>;
+
+    getSnykAppHost(): string;
+
+    getSnykApiVersion(): string;
+
+    getSnykIssuesApiVersion(): string;
+
+    getOrgSlug(orgId: string): Promise<string>;
+
+    isMocked(): boolean;
+
+    isAvailableInEntity(entity: Entity): boolean;
+
+    isShowResolvedInGraphs(entity: Entity): boolean;
+
+    getIssuesCount(issues: Array<Issue>): IssuesCount;
+
+    getIgnoredIssuesCount(issues: Array<Issue>): IssuesCount;
+}
+
+export class SnykApiClient implements SnykApi {
+    private readonly discoveryApi: DiscoveryApi;
+    private readonly proxyPathBase: string;
+    private readonly configApi: ConfigApi;
+    private readonly fetchApi: FetchApi;
+
+    private headers = {
+        "Content-Type": "application/json",
+        "User-Agent": "tech-services/backstage-plugin/1.0",
+    };
+
+    constructor(options: Options) {
+        this.discoveryApi = options.discoveryApi;
+        this.configApi = options.configApi;
+        this.proxyPathBase = options.proxyPathBase ?? DEFAULT_PROXY_PATH_BASE;
+        this.fetchApi = options.fetchApi;
+    }
+
+    private async getApiUrl() {
+        const baseUrl = await this.discoveryApi.getBaseUrl("proxy");
+        return `${baseUrl}${this.proxyPathBase}/snyk`;
+    }
+
+    private async fetch(
+        path: string,
+        method: string,
+        isV3 = false,
```

Comment:
I noticed that `true` is used five times in this file, while `false` is only used twice. It might be worth considering updating the default value based on this observation.